### PR TITLE
Update stac-browser to work with STAC 0.9 and 1.0.0-beta.1

### PR DESCRIPTION
**Pull request by lossyrob**
This PR updates stac-browser to work with 1.0.0-beta.1 and 0.9.0 STACs.

Changes of note:
- Refactor the `dictionary.json` property map to a js function for comment-ability; split the group definitions (extensions and data providers) from the properties. Added property definitions for 1.0.0-beta.1 extensions so they show up on the side bar - some exceptions for objects that would require more complicated display logic.
- Refactor validation logic - pull schemas directly from GitHub based on the STAC version of objects. Add a STAC_VERSION environment variable to default to in case STAC objects are older and do not have `stac_version` property.

Catalogs tested against:
- https://1tqdbvsut9.execute-api.us-west-2.amazonaws.com/v0/ (1.0.0-beta.1)
- https://storage.googleapis.com/pdd-stac/disasters/catalog.json (0.6)
- https://raw.githubusercontent.com/cholmes/sample-stac/master/stac/catalog.json (1.0.0-beta.1)
- https://geobase-spot-dev.s3.amazonaws.com (0.9.0)
- https://storage.googleapis.com/pdd-stac/disasters-0.9.0/catalog.json (0.9.0)
- https://geo.spacebel.be/stac/ (0.8.0)

Some catalogs had validation errors; I put in some workarounds to account for common mistakes, but in some cases the STACs will need to be corrected to work with the updated stac-browser.

### CODE REVIEW

File: src/util.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  if (!rsp.ok) {
+    throw new Error(`Loading error: ${rsp.statusText}`);
+  }
+
+  const schema = fixUpSchema(await rsp.json(), schemaUrl);
+
+  const loadSchema = async function(uri) {
+    let uriToFetch = modifyLoadSchemaUri(baseUrl, uri);
+
+    // Fetching the jsonschema causes a recursive loop; avoid that.
```
Comment (lossyrob) [inline src/util.js:60]: reminder: clarify comment

File: src/components/Catalog.vue [hunk 2]
```diff
@@ -595,11 +627,18 @@ export default {
         return null;
       }
 
+      // In STAC 0.8, temporal was changed from the direct array to an
+      // object with the property 'interval' containing an array of arrays.
+      // As a worrkaround, use the first element of that array.
```
Comment (lossyrob) [inline src/components/Catalog.vue:632]: reminder: fix typo

File: src/util.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  // Common Metadata
+  return `${baseUrl}/item-spec/json-schema/${uri}`;
+};
+
+const fixUpSchema = function(schema, schemaUri) {
+  // Make $id unique, otherwise AjV will complain
+  schema.$id = schemaUri + '#';
+
+  // Fix old schemas that have 'id' instead of '$id'
+  if("id" in schema) {
```
Comment (m-mohr) [inline src/util.js:32]: Do you really need to do that yourself or could you just set the ajv option `schemaId` to `auto`?

**Comment by lossyrob** [comment]
@m-mohr the two that I accounted for were:

 the spatial extent 'bbox' being a single array of coordinates instead of an array of arrays, 
https://github.com/radiantearth/stac-browser/pull/33/files#diff-372429037ae0197375dd49ce425801e3R603

and provider roles being a single string instead of an array.
https://github.com/radiantearth/stac-browser/pull/33/files#diff-372429037ae0197375dd49ce425801e3R474

File: src/util.js (hunk 3 above)
Comment (lossyrob) [reply src/util.js:33]: Hm, good call, but I tried this but was still getting an error

```
Uncaught (in promise) Error: schema $id is different from id
    at Ajv._get$IdOrId [as _getId] (ajv.js:398)
    at Ajv._addSchema (ajv.js:300)
    at async.js:31
```

Not sure why this is happening base on the Ajv docs; perhaps it's based on the secondary schemas loaded in that are somehow not affected by the options? Either way, I'll leave in this workaround for now.

**Comment by lossyrob** [comment]
Thank you for the review @m-mohr!

**Pull request merged**